module/PowerNSX.psm1

```diff
@@ -13969,6 +14001,174 @@ function Disable-NsxEdgeSsh {
+    for load balancer, site‐to‐site VPN, and NAT services.
+
+    The Set-NsxcliSettings cmdlet configure the cli Settings of the ESG
+    it is mandatory to specified the password...
+
+    .EXAMPLE
+    Get-NsxEdge Edge01 | Get-NsxcliSettings | Set-NsxCliSettings -password Vmware1!Vmware1!
+
+    Change the SSH Password
+
+    .EXAMPLE
+    Get-NsxEdge Edge01 | Get-NsxcliSettings | Set-NsxCliSettings -password Vmware1!Vmware1! -remoteAccess:$true
+
+    Enable the SSH on ESG (you can use also use Enable-NsxSSHEdgeSSH)
+
+    .EXAMPLE
+    Get-NsxEdge Edge01 | Get-NsxcliSettings | Set-NsxCliSettings -password Vmware1!Vmware1! -username powernsx
+
+    Set the SSH username to PowerNSX
+
+    .EXAMPLE
+    Get-NsxEdge Edge01 | Get-NsxcliSettings | Set-NsxCliSettings -password Vmware1!Vmware1! -sshLoginBannerText "My Login Banner"
+
+    Change the SSH Login Banner
+
+    .EXAMPLE
+    Get-NsxEdge Edge01 | Get-NsxcliSettings | Set-NsxCliSettings -password Vmware1!Vmware1! -passwordExpiry 30
+    Change the SSH Password Expiration to 30 (days)
+
+    #>
+
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSAvoidDefaultValueSwitchParameter","")] # Cant remove without breaking backward compatibility
+    param (
+
+        [Parameter (Mandatory=$true,ValueFromPipeline=$true,Position=1)]
+            [ValidateScript({ ValidateCliSettings $_ })]
+            [System.Xml.XmlElement]$cliSettings,
+        [Parameter (Mandatory=$false)]
+            [ValidateNotNullorEmpty()]
+            [String]$userName,
+        [Parameter (Mandatory=$true)]
+            [ValidateNotNullorEmpty()]
+            [String]$password,
+        [Parameter (Mandatory=$false)]
+            [ValidateNotNullorEmpty()]
+            [boolean]$remoteAccess,
+        [Parameter (Mandatory=$false)]
+            [ValidateNotNullorEmpty()]
+            [ValidateRange(1,99999)]
+            [int]$passwordExpiry,
+        [Parameter (Mandatory=$false)]
+            [ValidateNotNullorEmpty()]
+            [string]$sshLoginBannerText,
+        [Parameter (Mandatory=$False)]
+            #PowerNSX Connection object
+            [ValidateNotNullOrEmpty()]
+            [PSCustomObject]$Connection=$defaultNSXConnection
+
+    )
+
+    begin {
+
+    }
+
+    process {
+
+        #Create private xml element
+        $_cliSettings = $cliSettings.CloneNode($true)
+
+        #Store the edgeId and remove it from the XML as we need to post it...
+        $edgeId = $_cliSettings.edgeId
+        $_cliSettings.RemoveChild( $((Invoke-XPathQuery -QueryMethod SelectSingleNode -Node $_cliSettings -Query 'descendant::edgeId')) ) | out-null
+
+        #Using PSBoundParamters.ContainsKey lets us know if the user called us with a given parameter.
+        #If the user did not specify a given parameter, we dont want to modify from the existing value.
+
+        if ( $PsBoundParameters.ContainsKey('userName') ) {
+            $_cliSettings.username = $userName
```

Comment:
I also get nervous where I see assumption that a specific XML node exists.  Please get into habit of testing for and creating the element if it doesnt exist - i get bitten by this too often.